relay-profiling/src/sample.rs

```diff
@@ -95,6 +126,110 @@ impl Profile {
             frame.strip_pointer_authentication_code(addr);
         }
     }
+
+    fn remove_idle_samples_at_the_edge(&mut self) {
+        let mut active_ranges: HashMap<u64, Range<usize>> = HashMap::new();
+
+        for (i, sample) in self.samples.iter().enumerate() {
+            let is_active = match self.stacks.get(sample.stack_id) {
+                Some(stack) => !stack.is_empty(),
+                None => true,
+            };
+
+            if !is_active {
+                continue;
+            }
+
+            if let Some(range) = active_ranges.get_mut(&sample.thread_id) {
```

Comment:
Untested, but something like this should work:

```rs
active_ranges.entry(sample.thread).and_modify(|range| range.end += 1).or_insert(i..i+1)
```

Reply:
with small edit works :) thanks